configureWebpack.js

```diff
@@ -508,7 +508,7 @@ const extraPluginsProd = (terserOptions) => {
                 // Don't mangle class or function names as they may be used in error messages.
                 keep_classnames: true,
```

Comment:
are these 'keep_' flags relevant when 'mangle' is false?  I think we should remove for clarity, or turn into a comment

Reply:
One would think that they are not relevant when mangle is false. But I did not find that clearly stated in the terser documentation, neither here: https://github.com/terser/terser#minify-options nor https://github.com/terser/terser#mangle-properties-options.

One could assume from this line:
mangle (default true) — pass false to skip mangling names
in https://github.com/terser/terser#minify-options
that "names" includes function names and class names, but again that is not clearly stated.

Will have to search through the terser code or test our own code some more to be sure.

Reply:
I researched the various combinations in a client app.

The following three combinations all created the same sized 181kb un-mangled JS for the client app's application code.

```
keep_classnames: true,
keep_fnames: true,
mangle: false,
```

```
keep_classnames: false,
keep_fnames: false,
mangle: false,
```

```
// keep_classnames: true,
// keep_fnames: true,
mangle: false,
```

Switching to:
```
keep_classnames: false,
keep_fnames: false,
mangle: false,
```
caused the same application code to be reduced to 100kb, with all class and function names mangled.

I conclude that if we set mangle: false, we can leave keep_classnames and keep_fnames out of the config, because their values are overridden by mangle: false.

However, I do note that while we gain a lot in safety, we do lose a lot in terms of file compression, for application code.
In my earlier kb measurement mentioned in https://github.com/xh/hoist-react/issues/1429 
 not mangled "bundle size that was 9kb more: 617kb vs 608kb" I was reading the size of a file that contained mainly shared vendor code, which is mostly already minified by the vendor.

So, by not mangling anything, we gain a lot of safety, but we do get much  bigger files sizes for application specific code.

Maybe it is worth looking into if we can keep the original mangle settings:
```
keep_classnames: true,
keep_fnames: true,
mangle: true,
```

but have them only apply to non-vendor code.  We could exclude from mangling any chunks that start with vendor-

Reply:
I tested excluding files that start with "vendors~" from the terser process, but including other application code files.
The application code files were nicely compressed to 106kb, and the original highcharts error we are trying to avoid is avoided because vendor code is not "tersed".  But the vendors~ files are not minified at all between libs so we are leaving a lot of whitespace on the table in those files.  

Seems like the best option would be to run the terser plugin with :
```
keep_classnames: true,
keep_fnames: true,
mangle: true,
```
on the application files,
but with mangle: off on the vendors files.

I can explore doing this with a custom minify function.

Reply:
I've created a new branch that allows for mangling application code, but not mangling vendor code chunks, via a custom minify function: https://github.com/xh/hoist-dev-utils/compare/customTerserMinifyFn?expand=1

It has one drawback, so not sure it is a viable solution.

Reply:
Another option we could consider is just running the terser plugin twice.  First with exclude vendors~ and mangle: true
Second time with exclude !vendors and mangle: false.

Have to see if the negative match on exclude or include is possible, and if it really will run twice.

Reply:
I just tried this option:
"run the terser plugin twice. First with exclude vendors~ and mangle: true
Second time with include vendors and mangle: false."

It produces other bugs in the final output .

It seems like the last thing to try to get the smallest chunk sizes would be to turn off mangling of specifically those highcharts variables that are causing the issue.
This would probably work, but it might be a list that we have to update every so often.

Reply:
Putting this aside for now to look at more pressing issues.